iaso/models/base.py

```diff
@@ -906,9 +906,7 @@ class Instance(models.Model):
     device = models.ForeignKey("Device", null=True, blank=True, on_delete=models.DO_NOTHING)
     period = models.TextField(null=True, blank=True, db_index=True)
     entity = models.ForeignKey("Entity", null=True, blank=True, on_delete=models.DO_NOTHING, related_name="instances")
-    planning = models.ForeignKey(
-        "Planning", null=True, blank=True, on_delete=models.DO_NOTHING, related_name="instances"
-    )
+    planning = models.ForeignKey("Planning", null=True, blank=True, on_delete=models.SET_NULL, related_name="instances")
```

Comment:
This solves an `IntegrityError` ([trace.txt](https://github.com/BLSQ/iaso/files/13219140/trace.txt)) we encountered with @mbayangdiop95.

[My first try didn't work](https://github.com/BLSQ/iaso/pull/905/commits/65c7bd60a1b08533902578bf4422a3e453f6b287)…

Changing `on_delete=models.DO_NOTHING` to `on_delete=models.SET_NULL` fixed our problem.

I'm just wondering [if there was any business rule associated](https://github.com/BLSQ/iaso/pull/349/files#diff-c9377ad8ae865afaccd4478494b1f5d8608fe7a7e05740ba32315d7f0d88b6c2R816) with an explicit [`DO_NOTHING`](https://docs.djangoproject.com/en/4.2/ref/models/fields/#django.db.models.DO_NOTHING)?

Reply:
This looks ok to me and I think this will work well. I don't see any issue that this could create if the planning is gone from the db anyway.